Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -552,11 +552,18 @@ int RGroupDecomposition::add(const ROMol &inmol) {
     return -2;
   }
 
-  // in case the value ends up being changed in a future version of the code:
-  if (data->prunePermutations) {
-    data->permutationProduct = 1;
+  if (data->params.matchingStrategy != GA) {
+    size_t N = 1;
+    for (auto matche = data->matches.begin() + data->previousMatchSize;
+         matche != data->matches.end(); ++matche) {
+      size_t sz = matche->size();
+      N *= sz;
+    }
+    // oops, exponential is a pain
+    if (N * potentialMatches.size() > 100000) {
+      data->process(data->prunePermutations);
+    }
```

Comment:
I don't think you want to remove the `if (data->prunePermutations)` block, right? You just intend to add this piece back?

Reply:
@greglandrum That was actually intentional, because `data->permutationProduct` is unused. I also removed it from `RGroupDecompData` for the same reason. I have added comments about what was discussed. Also based on Brian’s advice, I have refrained from exposing `MAX_PERMUTATIONS` as a user-adjustable parameter.